src/components/form/GenericField/GenericField.tsx

```diff
@@ -185,19 +196,19 @@ class GenericField extends React.PureComponent<GenericFieldProps, GenericFieldSt
     /** deprecated. The GenericFieldEnum should be used directly instead */
     public static CUSTOM_TYPE: GenericFieldEnum.CUSTOM_TYPE = GenericFieldEnum.CUSTOM_TYPE;
 
-    static formatValue = (type: GenericFieldEnum, value: any): boolean | string[] | number => {
+    static formatValue = (type: GenericFieldEnum, value: boolean | string): boolean | string | string[] | number => {
         if (type === GenericFieldEnum.MULTI_SELECT_LIST_TYPE) {
-            return _.split(value, ',');
+            return _.split(String(value), ',');
         }
         if (type === GenericFieldEnum.BOOLEAN_TYPE) {
-            return (_.isBoolean(value) && value) || (_.isString(value) && value === 'true');
+            return (_.isBoolean(value) && value) || (_.isString(value) && String(value) === 'true');
```

Comment:
```suggestion
            return (_.isBoolean(value) && value) || (typeof value === 'string' && value === 'true');
```
IMO looks just a little bit more reasonable than converting string to string.

Reply:
Good point! I committed your suggestion.